cpp/src/arrow/compute/kernels/vector_nested.cc

```diff
@@ -76,6 +76,24 @@ struct ListParentIndicesArray {
 
   Status Visit(const LargeListType& type) { return VisitList(type); }
 
+  Status Visit(const FixedSizeListType& type) {
+    const int32_t slot_length = type.list_size();
+    const int64_t values_length = slot_length * (input->length - input->GetNullCount());
```

Comment:
Does `GetNullCount` guarantees the true number of nulls? After doing a grep in repo, `SetNullCount` is used scarcely as well as updates to `null_count` data member. I am just a bit confused on how the null count is updated.

Reply:
GetNullCount computes the nulls, yes - look at the implementation. If it were purely a getter it would be called `null_count`.

https://github.com/apache/arrow/blob/3bbec3f31ca16dd347b9ec88bd2b38fbf1a403b9/cpp/src/arrow/array/data.cc#L117-L129